src/js/view/controls/components/chapters.mixin.js

```diff
@@ -35,15 +39,29 @@ const ChaptersMixin = {
         if (Array.isArray(data)) {
             data.forEach((obj) => this.addCue(obj));
             this.drawCues();
+            const existingCues = this._model.get('cues');
+            const playerCues = existingCues.concat(data);
+            this._model.set('cues', playerCues);
         }
     },
 
     chaptersFailed: function () {},
 
     addCue: function (obj) {
+        this.cues.forEach((cue) => {
+            if (cue.time === obj.begin) {
+                this.removeCue(cue);
+            }
+        });
         this.cues.push(new Cue(obj.begin, obj.text));
     },
 
+    removeCue: function (cueToRemove) {
```

Comment:
This doesn't need to be its own function exposed on the object. Update the model, then reset and redraw all cues.

Reply:
> Fix a JS exception that was occurring when clearing out the queues using `jwplayer().setCues([{}]);`

Can't you just call `jwplayer().setCues([]);`? `{}` is not a valid cue. What calls it? Don't add a check to each cue, when you can handle this at a higher level and just reset.

> Prevent duplicate cues at the same time on the timeslider
Cues with the same time but different content are not duplicates. Establish rules about which types of cues take priority. Handle this when drawing the full list of cues not by removing and replacing elements one at a time.

Reply:
@robwalch 
The question I have would be how to tell what is adding cues. It just happens to be that chapters are added when `playlistItem` fires while the ad plugins add them when the playback is initiated. I assume that we would always want ad cues to take priority over chapter cues, right? If so, I can check in `drawCues()`.

This is a little stream of consciousness but I want to make sure I'm thinking along the same lines.